sdk/core/Azure.Core/src/Shared/DiagnosticScopeFactory.cs

```diff
@@ -85,15 +87,15 @@ public DiagnosticScope CreateScope(string name, System.Diagnostics.ActivityKind
         ///     result Azure.Storage.Blobs.BlobClient
         /// </summary>
 #if NETCOREAPP2_1
-        private static object? GetActivitySource(string ns, string name)
+        private object? GetActivitySource(string ns, string name)
 #else
-        private static ActivitySource? GetActivitySource(string ns, string name)
+        private ActivitySource? GetActivitySource(string ns, string name)
 #endif
         {
 #if NETCOREAPP2_1
-            if (!ActivityExtensions.SupportsActivitySource())
+            if (!_isStable && !ActivityExtensions.SupportsActivitySource())
```

Comment:
Is it correct that we don't want to return null now if `ActivityExtensions.SupportsActivitySource()` is `false` but `_isStable` is `true`?

Reply:
yes, `ActivityExtensions.SupportsActivitySource()` checks if users opted in experimental stuff. So we return `null` when instrumentation is neither stable, nor enabled.
I rewrote it slightly to be more readable.